lib/cadet/auth/providers/saml/nusstu_assertion_extractor.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Cadet.Auth.Providers.NusstuAssertionExtractor do
+  @moduledoc """
+  Extracts fields from NUS Student IdP SAML assertions.
+  """
+
+  @behaviour Cadet.Auth.Providers.AssertionExtractor
+
+  def get_username(assertion) do
+    Map.get(assertion.attributes, "samaccountname")
+  end
+
+  def get_name(assertion) do
```

Comment:
Will require IdP to provide a name attribute within the assertion, as it is used in our application.

Reply:
Yup, I think it's good for testing for now though, since we have no information about this.